src/modules.js

```diff
@@ -215,25 +215,18 @@ var LibraryManager = {
         if (lib[target + '__asm']) continue; // This is an alias of an asm library function. Also needs to be fully optimized.
         if (!isNaN(target)) continue; // This is a number, and so cannot be an alias target.
         if (typeof lib[target] === 'undefined' || typeof lib[target] === 'function') {
-          // If the alias provides a signature, then construct a specific 'function foo(a0, a1, a2) { [return] _target(a0, a1, a2); }' form of forwarding.
-          // Otherwise construct a generic 'function foo() { return _target.apply(null, arguments); }' forwarding.
-          // The benefit of the first form is that Closure is able to fully inline and reason about the function.
-          // Note that the signature is checked on the alias function, not on the target function. That allows aliases to choose individually which form
-          // to use. 
-          if (lib[x + '__sig']) {
-            var argCount = lib[x + '__sig'].length - 1;
-            var ret = lib[x + '__sig'] == 'v' ? '' : 'return ';
-            var args = genArgSequence(argCount).join(',');
-            lib[x] = new Function(args, ret + '_' + target + '(' + args + ');');
-          } else {
-            // The alias we generate uses `arguments` to forward the call. If there is no explicit documentation for the alias, mark
-            // it variadic for Closure type checking to know.
-            var docs = lib[x + '__docs'];
-            if (!docs || (docs.indexOf('@type') == -1 && docs.indexOf('@param') == -1)) {
-              lib[x + '__docs'] = (docs || '') + '/** @type {function(...*):?} */';
-            }
-            lib[x] = new Function('return _' + target + '.apply(null, arguments)');
+          // When functions are aliased, the alias target must provide a signature for the function so that an efficient form of forwarding can be implemented.
+          // Primarily read the signature on the alias, and secondarily on the target.
+          var sig = lib[x + '__sig'] || lib[target + '__sig'];
```

Comment:
Error here is they disagree? (And perhaps even error if _sig is redundantly set on the alias when it already exists on the target?  . .since that would be extra noise that would make the code more brittle.)

Reply:
It can be useful for the signatures to differ e.g. for optional parameters. Though perhaps can add support for that when a case arises.